packages/java/engine-core/src/main/java/dev/hilla/engine/ParserConfiguration.java

```diff
@@ -0,0 +1,184 @@
+package dev.hilla.engine;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.Set;
+
+import javax.annotation.Nonnull;
+
+import dev.hilla.parser.core.PluginConfiguration;
+import dev.hilla.parser.plugins.backbone.BackbonePlugin;
+import dev.hilla.parser.plugins.model.ModelPlugin;
+import dev.hilla.parser.plugins.nonnull.NonnullPlugin;
+import dev.hilla.parser.plugins.transfertypes.TransferTypesPlugin;
+import dev.hilla.parser.utils.ConfigList;
+
+public final class ParserConfiguration {
+    private String endpointAnnotation;
+    private String endpointExposedAnnotation;
+    private String openAPIBasePath;
+    private Plugins plugins;
+
+    public Optional<String> getEndpointAnnotation() {
+        return Optional.ofNullable(endpointAnnotation);
+    }
+
+    public Optional<String> getEndpointExposedAnnotation() {
+        return Optional.ofNullable(endpointExposedAnnotation);
+    }
+
+    public Optional<String> getOpenAPIBasePath() {
+        return Optional.ofNullable(openAPIBasePath);
+    }
+
+    public Optional<Plugins> getPlugins() {
+        return Optional.ofNullable(plugins);
+    }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) {
+            return true;
+        }
+        if (o == null || getClass() != o.getClass()) {
+            return false;
+        }
+        var that = (ParserConfiguration) o;
+        return Objects.equals(endpointAnnotation, that.endpointAnnotation)
+                && Objects.equals(endpointExposedAnnotation,
+                        that.endpointExposedAnnotation)
+                && Objects.equals(openAPIBasePath, that.openAPIBasePath)
+                && Objects.equals(plugins, that.plugins);
+    }
+
+    @Override
+    public int hashCode() {
+        return Objects.hash(endpointAnnotation, endpointExposedAnnotation,
+                openAPIBasePath, plugins);
+    }
+
+    void setEndpointAnnotation(String endpointAnnotation) {
+        this.endpointAnnotation = endpointAnnotation;
+    }
+
+    void setEndpointExposedAnnotation(String endpointExposedAnnotation) {
+        this.endpointExposedAnnotation = endpointExposedAnnotation;
+    }
+
+    void setOpenAPIBasePath(String openAPIBasePath) {
+        this.openAPIBasePath = openAPIBasePath;
+    }
+
+    void setPlugins(Plugins plugins) {
+        this.plugins = plugins;
+    }
+
+    public static class Plugin {
```

Comment:
Could we re-use `Plugin` and `Plugins` classes for both `ParserConfiguration` and `GeneratorConfiguration`?